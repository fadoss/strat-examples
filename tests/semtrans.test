metalevel/semtrans-test

red run('ab.RuleApp) .
red run('st.StratCall) .
red run('match_s.t._['X:Thing, '_=_['X:Thing, 'b.Thing]], 'c.Thing) .
red run('amatch_s.t._['b.Thing, 'trueC.EqCondition], 'f['b.Thing, 'c.Thing]) .
red run('_?_:_['ab.RuleApp,'bc.RuleApp,'ac.RuleApp]) .
red run('matchrew['f['X:Thing, 'b.Thing], 'trueC.EqCondition, '_using_['X:Thing, 'ab.RuleApp]], 'f['a.Thing, 'b.Thing]) .
red run('matchrew['f['X:Thing, 'Y:Thing], 'trueC.EqCondition, '_`,_['_using_['X:Thing, 'ab.RuleApp], '_using_['Y:Thing, 'bc.RuleApp]]], 'f['a.Thing, 'b.Thing]) .
red run('matchrew['f['X:Thing, 'Y:Thing], '_=_['X:Thing, 'Y:Thing], '_`,_['_using_['X:Thing, 'ab.RuleApp], '_using_['Y:Thing, 'idle.Strat]]], 'f['a.Thing, 'a.Thing]) .
