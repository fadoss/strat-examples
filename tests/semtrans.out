==========================================
reduce in REWSEM-TEST : run('ab.RuleApp) .
rewrites: 878
result ResultPair: {'sol['b.Thing], 'Task}
==========================================
reduce in REWSEM-TEST : run('st.StratCall) .
rewrites: 812
result ResultPair: {'sol['b.Thing], 'Task}
==========================================
reduce in REWSEM-TEST : run('match_s.t._['X:Thing, '_=_['X:Thing, 'b.Thing]],
    'c.Thing) .
rewrites: 750
result ResultPair: {'none.Tasks, 'Tasks}
==========================================
reduce in REWSEM-TEST : run('amatch_s.t._['b.Thing, 'trueC.EqCondition], 'f[
    'b.Thing, 'c.Thing]) .
rewrites: 770
result ResultPair: {'sol['f['b.Thing, 'c.Thing]], 'Task}
==========================================
reduce in REWSEM-TEST : run('_?_:_['ab.RuleApp, 'bc.RuleApp, 'ac.RuleApp]) .
rewrites: 826
result ResultPair: {'sol['c.Thing], 'Task}
==========================================
reduce in REWSEM-TEST : run('matchrew['f['X:Thing, 'b.Thing],
    'trueC.EqCondition, '_using_['X:Thing, 'ab.RuleApp]], 'f['a.Thing,
    'b.Thing]) .
rewrites: 832
result ResultPair: {'sol['f['b.Thing, 'b.Thing]], 'Task}
==========================================
reduce in REWSEM-TEST : run('matchrew['f['X:Thing, 'Y:Thing],
    'trueC.EqCondition, '_`,_['_using_['X:Thing, 'ab.RuleApp], '_using_[
    'Y:Thing, 'bc.RuleApp]]], 'f['a.Thing, 'b.Thing]) .
rewrites: 900
result ResultPair: {'sol['f['b.Thing, 'c.Thing]], 'Task}
==========================================
reduce in REWSEM-TEST : run('matchrew['f['X:Thing, 'Y:Thing], '_=_['X:Thing,
    'Y:Thing], '_`,_['_using_['X:Thing, 'ab.RuleApp], '_using_['Y:Thing,
    'idle.Strat]]], 'f['a.Thing, 'a.Thing]) .
rewrites: 860
result ResultPair: {'sol['f['b.Thing, 'a.Thing]], 'Task}
Bye.
